parser-typechecker/src/Unison/Codebase/Path.hs

```diff
@@ -189,11 +196,11 @@ fromList = Path . Seq.fromList
 ancestors :: Absolute -> Seq Absolute
 ancestors (Absolute (Path segments)) = Absolute . Path <$> Seq.inits segments
 
-hqSplitFromName' :: Name -> Maybe HQSplit'
-hqSplitFromName' = fmap (fmap HQ'.fromName) . Lens.unsnoc . fromName'
+hqSplitFromName' :: Name -> HQSplit'
+hqSplitFromName' = fmap HQ'.fromName . fromJust . Lens.unsnoc . fromName'
 
-splitFromName :: Name -> Maybe Split
-splitFromName = unsnoc . fromName
+splitFromName :: Name -> Split
+splitFromName = fromJust . unsnoc . fromName
```

Comment:
Sure, `error` would be ok. I don't love how it makes the code dirtier for an impossible case. Would perhaps be better to make `Name.segments` return a non-empty list, so that unconsing it isn't partial. Maybe we should do that instead?

Reply:
I like non-empty lists